packages/block-library/src/gallery/edit.js

```diff
@@ -460,6 +470,14 @@ function GalleryEdit( props ) {
 
 	const hasLinkTo = linkTo && linkTo !== 'none';
 
+	const gap = attributes.style?.spacing?.blockGap
+		? `.wp-container-${ id } { --wp--style--unstable-gallery-gap: ${ attributes.style.spacing.blockGap } }`
```

Comment:
I'm not sure if what I'm seeing is a bug or not, but the class – or more specifically, the id – generated here does not correspond to any element in the DOM.

<img width="709" alt="Screen Shot 2022-01-25 at 12 33 38 pm" src="https://user-images.githubusercontent.com/6458278/150894201-13a05770-01c0-44be-91a3-eba3d5067808.png">

I can reproduce it by adding a couple of blocks above the gallery in the editor.

Do we still need to add the class to the container?

Also, do you think it should match the one added in PHP, e.g., `wp-block-gallery-`

The frontend works fine, but because of the above the editor and frontend look different to me:

| Editor | Frontend |
| ------------- | ------------- |
| <img width="400" alt="Screen Shot 2022-01-25 at 12 35 35 pm" src="https://user-images.githubusercontent.com/6458278/150894377-12cd5aeb-2ff0-4ec2-a1ae-98677e9d6585.png">  | <img width="400" alt="Screen Shot 2022-01-25 at 12 35 30 pm" src="https://user-images.githubusercontent.com/6458278/150894392-b645fc14-4f9e-440a-a3f6-04c740b77a28.png">  |

Reply:
ah, thanks, I have misunderstood how `useInstanceId` works - worked perfectly when I only had one block on the page, will take another look at it

Reply:
@ramonjd Give it another try now - rather than having to add an additional class I just went with using the existing block id as in the [original commit](https://github.com/WordPress/gutenberg/pull/38164/commits/957fd862a0f0e256e66e2d2a4f463e45b24c02ef#diff-7be1df22c250a7378a0796b6c68da7fc1178a98d9e88ab9f59868a6d0b8c467fR463) - let me know if you can see any issues with this approach.